src/rust/generic-subgraph-generator/src/models/network/inbound.rs

```diff
@@ -0,0 +1,139 @@
+use graph_descriptions::graph_description::*;
+use graph_descriptions::process::ProcessState;
+use graph_descriptions::process_inbound_connection::ProcessInboundConnectionState;
+use serde::{Deserialize, Serialize};
+
+use graph_descriptions::network_connection::NetworkConnectionState;
+use graph_descriptions::node::NodeT;
+use std::convert::TryFrom;
+
+// In an inbound connection "src" is where the connection is coming from
+#[derive(Clone, Debug, Hash, Serialize, Deserialize)]
+pub struct ProcessInboundConnectionLog {
+    /// The pid of the process receiving the connection
+    pid: u64,
+    src_ip_addr: String,
+    src_port: u32,
+    dst_port: u32,
+    dst_hostname: String,
+    dst_ip_addr: String,
+    protocol: String,
+    timestamp: u64,
+}
+
+impl TryFrom<ProcessInboundConnectionLog> for Graph {
+    type Error = String;
+
+    fn try_from(conn_log: ProcessInboundConnectionLog) -> Result<Self, Self::Error> {
+        let mut graph = Graph::new(conn_log.timestamp);
+
+        let asset = AssetBuilder::default()
+            .asset_id(conn_log.dst_hostname.clone())
+            .hostname(conn_log.dst_hostname.clone())
+            .build()?;
+
+        // A process creates an outbound connection to dst_port
+        let process = ProcessBuilder::default()
+            .asset_id(conn_log.dst_hostname.clone())
+            .state(ProcessState::Existing)
+            .process_id(conn_log.pid)
+            .last_seen_timestamp(conn_log.timestamp)
+            .build()?;
+
+        let inbound = ProcessInboundConnectionBuilder::default()
+            .asset_id(conn_log.dst_hostname.clone())
+            .state(ProcessInboundConnectionState::Existing)
+            .ip_address(conn_log.dst_ip_addr.clone())
+            .protocol(conn_log.protocol.clone())
+            .port(conn_log.dst_port)
+            .created_timestamp(conn_log.timestamp)
+            .build()?;
+
+        let dst_ip = IpAddressBuilder::default()
+            .ip_address(conn_log.dst_ip_addr.clone())
+            .last_seen_timestamp(conn_log.timestamp)
+            .build()?;
+
+        let src_ip = IpAddressBuilder::default()
+            .ip_address(conn_log.src_ip_addr.clone())
+            .last_seen_timestamp(conn_log.timestamp)
+            .build()?;
+
+        let src_port = IpPortBuilder::default()
+            .ip_address(conn_log.src_ip_addr.clone())
+            .protocol(conn_log.protocol.clone())
```

Comment:
As it turns out, we were missing the `.protocol()` line here as caught by unit testing :D